src/comments.js

```diff
@@ -480,6 +480,13 @@ function handleIfStatementComments(
     return true;
   }
 
+  // For `if (a) /* comment */ {}`,
+  // we look at the previous character to see if it is a ).
+  if (getPreviousNonSpaceCharacter(text, comment) === ")") {
```

Comment:
Why not look at the next character and see if it's a `{` ?

Reply:
If I check `{`, it would still miss cases such as 

```
if (typeof err === "string") // comment
    callback(err);
```

though.